lib/rules/no-console.js

```diff
@@ -40,26 +102,52 @@ module.exports = {
         return {
 
             MemberExpression: function(node) {
+                var scope = context.getScope();
+                var passedProperty = getPropertyName(node);
+                var passedMemberExpressionProperty = getPropertyName(node.object);
+                var variable;
+                var propertyIsAllowed;
 
-                if (node.object.name === "console") {
-                    var blockConsole = true;
+                if (context.options.length > 0 && context.options[0].allow) {
+                    propertyIsAllowed = (context.options[0].allow.indexOf(passedProperty) > -1);
+                }
 
-                    if (context.options.length > 0) {
-                        var allowedProperties = context.options[0].allow;
-                        var passedProperty = node.property.name;
-                        var propertyIsAllowed = (allowedProperties.indexOf(passedProperty) > -1);
+                // checks console.*() [* means any console method]
+                if (node.object.name === "console") {
+                    variable = astUtils.getVariableByName(scope, "console");
 
-                        if (propertyIsAllowed) {
-                            blockConsole = false;
-                        }
+                    if (!isShadowed(variable) && !propertyIsAllowed) {
+                        report(context, node);
                     }
 
-                    if (blockConsole) {
-                        context.report(node, "Unexpected console statement.");
+                // checks window.console.*(), self.console.*(), && this.console.*()
+                } else if (passedMemberExpressionProperty === "console") {
+                    if (context.options.length && context.options[0].disallowGlobals) {
+                        var passedObject = node.object.object;
+                        var disallowedGlobals = context.options[0].disallowGlobals;
+
+                        switch (passedObject.name) {
+                            case "window":
```

Comment:
We only want to flag `window.console.log`, but not `myobject.console.log`, so we should hardcode something. I'm not 100% sure about `self` and `this` though. Also, if we do want to flag `self` those two cases should be collapsed, since they are doing the same thing.